lib/ArduinoNative/src/WString.h

```diff
@@ -290,7 +291,27 @@ class String
 
                 delete[] m_buffer;
                 m_buffer = tmp;
-                m_size += str.m_size;
+                m_size += str.m_size - 1;
+            }
+        }
+
+        return *this;
+    }
+
+    String& operator+=(const char* cstr)
+    {
+        if (nullptr != cstr)
+        {
+            char* tmp = new char[m_size + strlen(cstr)];
+
+            if (nullptr != tmp)
+            {
+                strcpy(tmp, m_buffer);
+                strcat(tmp, cstr);
+
+                delete[] m_buffer;
+                m_buffer = tmp;
+                m_size   = m_size + strlen(cstr);

```

Comment:
repetitive use of strlen(cstr) - performance !

Better 
size_t len = (nullptr != cstr) ? strlen(cstr) : 0U;

if  (0U != len) 
{ 
  .....